File: interactor/templates/app-java/src/main/java/interactor/_InteractorImpl.java
package <%= appPackage %>.domain.interactors.<%= interactorPackageName %>;

import rx.Scheduler;
import <%= appPackage %>.domain.interactors.base.BaseInteractor;

import rx.Observable

public class <%= interactorName %>InteractorImpl extends BaseInteractor implements <%= interactorName %>Interactor {

    public <%= interactorName %>InteractorImpl(Scheduler executor) {
        super(executor);
    }

    public Observable<Object> invoke() {
        return Observable.just(null);
    }

}

File: interactor/templates/app-kotlin/src/main/java/interactor/_InteractorImpl.kt
package <%= appPackage %>.domain.interactors.<%= interactorPackageName %>

import io.reactivex.Scheduler
import <%= appPackage %>.domain.interactors.base.BaseInteractor

import io.reactivex.Observable

class <%= interactorName %>InteractorImpl (executor: Scheduler) : BaseInteractor(executor), <%= interactorName %>Interactor {

    fun invoke(): Observable<Any> {
        return Observable.just(null)
    }

}
